### PR TITLE
More abstraction of AB test code

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This further encapsulates our AB testing logic into one place

## Why?
Because we want to be able to setup AB testing for different islands independently

### CODE REVIEW

File: dotcom-rendering/src/web/components/WithABProvider.tsx [hunk 1]
```diff
@@ -1,31 +1,59 @@
-import { CoreAPIConfig, OphanAPIConfig } from '@guardian/ab-core/dist/types';
-import { ABProvider } from '@guardian/ab-react';
+import { useEffect } from 'react';
+import { CoreAPIConfig } from '@guardian/ab-core/dist/types';
+import { ABProvider, useAB } from '@guardian/ab-react';
+import { getCookie, log } from '@guardian/libs';
+import { tests } from '@frontend/web/experiments/ab-tests';
```
Comment (sndrs) [inline dotcom-rendering/src/web/components/WithABProvider.tsx:5]:
```suggestion
import { tests } from '../experiments/ab-tests';
```
in prep again for #3936

**Review by sndrs — CHANGES_REQUESTED** [review]
couple of places to use relative paths instead of aliases if that's ok (and i've got them right?)  in prep for #3936

File: dotcom-rendering/src/web/components/App.tsx [hunk 2]
```diff
@@ -22,6 +20,7 @@ import { decideTheme } from '@root/src/web/lib/decideTheme';
 import { decideDisplay } from '@root/src/web/lib/decideDisplay';
 import { decideDesign } from '@root/src/web/lib/decideDesign';
 import { useOnce } from '@root/src/web/lib/useOnce';
+import { getOphanRecordFunction } from '@root/src/web/browser/ophan/ophan';
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/App.tsx:23]: Thanks!

**Review by sndrs — APPROVED** [review]
lovely

**Pull request merged**